src/exeptions/api.error.js

```diff
@@ -0,0 +1,32 @@
+export class ApiError extends Error {
+  constructor({message, status, errors = {}}) {
+    super(message);
+
+    this.status = status;
+    this.errors = errors;
+  }
+
+  static badRequest(message, errors) {
+    return new ApiError({
+      message,
+      errors,
+      status: 400,
+    });
+  }
+
+  static unAuthorized(errors) {
```

Comment:
The `errors` parameter in the `unAuthorized` method should be optional, similar to how it's handled in the constructor. Consider providing a default value, such as an empty object, to avoid potential issues when no errors are passed.